src/index.js

```diff
@@ -105,6 +111,12 @@ class File {
   }
 
   async addSchema() {
+    if (this.displayName === 'FileInterface') {
+      // tested with big file 1.9Gb 
+      this.descriptor.data = await browser.readCSV(this.descriptor); 
```

Comment:
I guess for now having this is just only for CSV is enough, but should we have `addSchema()` for the other formats too?
cc: @anuveyatsu

Reply:
Ideally, it should work for CSV and XLSX but it should be ok for now.